### PR TITLE
Compute noise covariance on resting-state recording

### PR DESCRIPTION
**Pull request by hoechenberger**
We already support "custom" noise covariances, but it seems to become necessary to explicitly allow for resting-state data as a source of noise.

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: scripts/preprocessing/_01_maxfilter.py [hunk 1]
```diff
@@ -124,54 +124,93 @@ def run_maxwell_filter(*, cfg, subject, session=None, run=None):
...
+        )
+
+        # Perform a sanity check: empty-room rank should exactly match the
+        # experimental data rank after Maxwell filtering; resting-state rank
+        # should be eauql or be greater than experimental data rank.
+        #
+        # We're treating the two cases differently, because we don't
+        # copy the bad channel selection from the reference run over to
+        # the resting-state recording.
+        # TODO: This decision is open for discussion!
```
Comment (hoechenberger) [inline scripts/preprocessing/_01_maxfilter.py:176]:
```suggestion
```

File: scripts/source/_03_make_cov.py [hunk 2]
```diff
@@ -61,35 +61,53 @@ def compute_cov_from_epochs(cfg, subject, session, tmin, tmax):
...
-    raw_er = mne.io.read_raw_fif(raw_er_fname, preload=True)
-    cov = mne.compute_raw_covariance(raw_er, method='shrunk', rank='info')
+    raw_noise = mne.io.read_raw_fif(bids_path_raw_noise, preload=True)
+
+    bids_path_epochs = bids_path_raw_noise.copy().update(
+        task=cfg.task,
+        # processing='clean',
+        processing=None,
+        suffix='epo'
+    )
```
Comment (hoechenberger) [inline scripts/source/_03_make_cov.py:109]:
```suggestion
```

**Comment by hoechenberger** [comment]
@agramfort It's behaving like expected and I get good results with this on the TIWM data! Please feel free to merge, or let's discuss if you're not happy with the approach and/or implementation

File: config.py [hunk 3]
```diff
@@ -1616,11 +1616,15 @@ def mri_landmarks_kind(bids_path):
...
+If ``'emptyroom'``, the noise covariance matrix will be estimated from an
 empty-room MEG recording. The empty-room recording will be automatically
 selected based on recording date and time.
 
-If ``ad-hoc``, the a diagonal ad-hoc noise covariance matrix will be used.
+If ``''rest'``, the noise covariance will be estimated from a resting-state
+recording (i.e., a recording with `task-rest` and without a `run` in the
+filename).
+
+If ``'ad-hoc'``, the a diagonal ad-hoc noise covariance matrix will be used.
```
Comment (agramfort) [inline config.py:1627]:
```suggestion
If ``'ad-hoc'``, a diagonal ad-hoc noise covariance matrix will be used.
```

File: scripts/preprocessing/_01_maxfilter.py [hunk 4]
```diff
@@ -124,54 +124,92 @@ def run_maxwell_filter(*, cfg, subject, session=None, run=None):
...
                                      run=run))
-        raw_er_sss = mne.preprocessing.maxwell_filter(raw_er,
-                                                      **common_mf_kws)
+        raw_noise_sss = mne.preprocessing.maxwell_filter(
+            raw_noise, **common_mf_kws
+        )
+
+        # Perform a sanity check: empty-room rank should exactly match the
+        # experimental data rank after Maxwell filtering; resting-state rank
+        # should be eauql or be greater than experimental data rank.
```
Comment (agramfort) [inline scripts/preprocessing/_01_maxfilter.py:171]:
```suggestion
        # should be equal or be greater than experimental data rank.
```

**Review by agramfort — APPROVED** [review]
Just 2 typos

🙏 @hoechenberger

**Pull request merged**